mtp_api/apps/transaction/views.py

```diff
@@ -32,8 +34,24 @@ class Meta:
         fields = ['upload_counter']
 
 
+class OwnPrisonListModelMixin(object):
+
+    def get_prison_set_for_user(self):
+        try:
+            return PrisonUserMapping.objects.get(user=self.request.user).prisons.all()
+        except PrisonUserMapping.DoesNotExist:
+            return Prison.objects.none()
+
+    def get_queryset(self):
+        qs = super(OwnPrisonListModelMixin, self).get_queryset()
+        if self.request.user.is_superuser:
```

Comment:
I read the TODO comment on the related test and I agree. Maybe it's worth deleting this `is_superuser` check and add it later on. Otherwise we might forget to add a testcase for this.